projects/omang-service-mediator/src/modules/mpi/services/mpi.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { AxiosRequestConfig } from 'axios';
+import { Agent } from 'https';
+import { config } from 'src/config';
+import { Bundle } from 'fhir/r4';
+
+@Injectable()
+export class MasterPatientIndex {
+  private readonly logger = new Logger(MasterPatientIndex.name);
+  private readonly clientRegistryUrl: string;
+  private readonly authHeader: string;
+  private readonly devMode: boolean;
+
+  constructor(private readonly httpService: HttpService) {
+    this.clientRegistryUrl = config.get('ClientRegistry:ApiUrl');
+    const authString = ''; //`${client}:${password}`;
+    this.authHeader = `Basic ${Buffer.from(authString).toString('base64')}`;
+    this.devMode = config.get('ClientRegistry:devMode') === 'true';
+  }
+
+  private getHttpOptions(): AxiosRequestConfig {
+    const options: AxiosRequestConfig = {
+      headers: {
+        Authorization: this.authHeader,
+        'Content-Type': 'application/fhir+json',
+      },
+    };
+
+    if (this.devMode) {
+      options.httpsAgent = new Agent({
+        rejectUnauthorized: false,
+      });
+    }
+
+    return options;
+  }
+
+  async getSearchBundle(query): Promise<any> {
+    try {
+      const searchResponse = await this.httpService.axiosRef.get<Bundle>(
+        `${this.clientRegistryUrl}/Patient`,
+        {
+          params: { identifier: query },
+          headers: {
+            'Content-Type': 'application/fhir+json',
+            'x-openhim-clientid': 'OmangSvc'
+          },
+        },
+      );
+
+      return searchResponse.data;
+    } catch (error) {
+      this.logger.error(
+        'Could not get CR bundle for patient with ID ' +
+          query +
+          '\n' +
+          error,
```

Comment:
Similar to a previous comment, to keep consistency and make use of template literals to inject the variables instead of string concatenation.